img_proof/ipa_azure.py

```diff
@@ -193,6 +214,30 @@ def _create_storage_profile(self):
                     'create_option': 'FromImage'
                 }
             }
+        elif self.gallery_name:
+            try:
+                image = self.compute.gallery_image_versions.get(
+                    self.gallery_resource_group,
+                    self.gallery_name,
+                    self.image_id,
+                    self.image_version
+                )
+                image_id = image.id
+            except Exception:
+                raise AzureCloudException(
+                    'Image with name: {0} and version: {1} '
+                    'not found in gallery: {2}.'.format(
+                        self.image_id,
+                        self.image_version,
+                        self.gallery_name
```

Comment:
I think the message should include the `gallery_resource_group`. It is conceivable that a user may have an A/B setup where the `gallery_name` is the same but in different resource groups, one for testing and one for production.